src/components/calendar/helpers.ts

```diff
@@ -71,6 +76,33 @@ export function convertToDate(value?: Date | string | null): Date | null {
   return isString(value) ? parseISODate(value) : isValidDate(value);
 }
 
+/**
+ * Converts the given value to a DateRangeValue object.
+ *
+ * If the value is already a valid DateRangeValue object, it is returned directly.
+ * If the value is a string, it is parsed to object and returned if it fields are valid dates.
+ * If the value is null or undefined, null is returned.
+ * If the parsing fails, null is returned.
+ */
+export function convertToDateRange(
+  value?: DateRangeValue | string | null
+): DateRangeValue | null {
+  if (!value) {
+    return null;
+  }
+
+  if (isString(value)) {
+    const obj = JSON.parse(value);
+    const start = isValidDate(new Date(obj.start));
+    const end = isValidDate(new Date(obj.end));
```

Comment:
```suggestion
    const start = convertToDate(obj.start);
    const end = convertToDate(obj.end);
```
I would like to keep accepting only valid ISO 8601 formatted strings.